### DISCUSSION TITLE
Homebrew is dead for macOS Catalina and below.

### DISCUSSION THREAD

**Discussion by euguk007** [Homebrew/discussions#4216] [Everyday usage]
### Output of `brew config`

```shell
Cmake, Aom and Doxygen are not installable
These are required for PHP and MariaDB
```

### Output of `brew doctor`

```shell
Warning: You are using macOS 10.15.
We (and Apple) do not provide support for this old version.
```

### Description of issue

At least, make previous formuli bottles to be available for downloads.
What used to be a decent add-on to a Mac grew into a non-backward compatible software.

**Comment by SMillerDev** [comment]
> At least, make previous formuli bottles to be available for downloads.

They are, php and mariadb depend on newer bottles though.

**Reply by euguk007** [reply]
This is exactly the case. My surprise is that Homebrew does not offer an option for older macOS to use older bottles, which would do the required job.

**Reply by euguk007** [reply]
The funny thing is that I only need the 'dead' Homebrew to install Laravel Valet :)
And since Laravel Valet is hooked to Homebrew like Homebrew is hooked to Apple - I can't use it :)
Silly really :)

**Reply by euguk007** [reply]
Just to make my point bolder :)

I also run the latest Lubuntu as a dual boot on my MacBook Pro that runs Catalina.

I installed Homebrew Linux version on the same hardware that runs macOS Catalina on my second SSD.
Homebrew runs perfectly on my Mac Linux. No problems with any latest upgrades of any bottles :)))

What is really strange is that in Linux, when I install PHP of any kind, it does not require a dependancy of Doxygen :)))
What the heck?! Why does the PHP of Homebrew for Mac requires it?!

Cmake is installed easily on the same MacBook Pro under the Linux version of Homebrew.

And, if you consider the fact that Homebrew was designed as a substitute of Linux capabilities for Apple machines in the first place, the fact that Homebrew dismisses Mac machines of the past is a very peculiar behaviour :)))

It makes no sense whatsoever!

**Reply by SMillerDev** [reply]
> the fact that Homebrew dismisses Mac machines of the past is a very peculiar behaviour :)))

> It makes no sense whatsoever!

It makes a lot of sense. Homebrew only works with software that the developers support. That goes for PHP (we dropped 7.4 recently for that reason) and it also goes for macOS and Linux versions. The latest lubuntu is supported by its developers so Homebrew will run on it. Catalina is unsupported by Apple so Homebrew will not run on it.

---

It seems like the behaviour you want is more in line with what macports provides, which afaik is "we'll ship it if it builds"

**Reply by euguk007** [reply]
@SMillerDev - I shall agree to disagree. Homebrew is a friendly solution for selected users. And an unfriendly piece of software for the rest. It is dead for me, and (I am sure) for many others. Good luck for the future of Homebrew, and thanks for your responses :)
Cheers!

**Reply by Nllii** [reply]
You can do the following to revert it back and then install what you wanted, then revert it back again to master or main.

Each time something is not supported on Catalina: call this script. 

```bash 
#!/bin/bash

#TODO: find all hash supported on Catalina and add them to the array. 
COMMIT_HASH=$1
if [ -z "$COMMIT_HASH" ]; then
    echo -e "- use git log(not practical) \nvisit https://github.com/Homebrew/homebrew-core/search?q=php&type=commits)"
    echo ""
    echo ""
    printf "provide a commit hash example:f54d39e63b609a2ae25dd039f9edae6c82120f96 not 4ea73eb33aa: "
    read COMMIT_HASH

fi
cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/
git reset --hard $COMMIT_HASH
brew update # this is just incase homebrew had a warning about the commit hash that was just reset to.
# verify you have changed HEAD: 
brew config 
# Core tap origin: https://github.com/Homebrew/homebrew-core
# Core tap HEAD: f54d39e63b609a2ae25dd039f9edae6c82120f96
# Core tap last commit: 3 months ago
# Core tap branch: master

echo "Reverted to commit hash $COMMIT_HASH on $(date)" >> scripts.log
# revert back to master or main.
git checkout master
# finally git pull to last commit again to revert back to current:
cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/
git pull 
# HEAD: ba6d87eed9ce31f42fa2b33a4e6067d9cf17f13f
# Last commit: 29 hours ago
# Core tap origin: https://github.com/Homebrew/homebrew-core
# Core tap HEAD: 9f6f46344e65033096b2ba68001ec38467e84a8f
# Core tap last commit: 36 minutes ago
# Core tap branch: master
```

Now you can do ```brew install php```  for Catalina  again. 

The goal is to find all the Formula that was support for Catalina in the commit history. 
https://github.com/Homebrew/homebrew-core/search?q=php&type=commits

**Reply by SMillerDev** [reply]
If you're going to manually install the latest available version for Catalina you might as well just stop updating. It'll save you a step.

**Reply by euguk007** [reply]
@Nllii - Thank you! I will try this script on my Catalina :)

@SMillerDev - If there is a way to manually install all required formula for Catalina to be able to run Laravel Valet I would stop updating those bottles :)

So far, I found a solution like so :)

![Screenshot](https://user-images.githubusercontent.com/83634871/221370547-fd138abd-04e4-4c03-b58a-ada7b740599b.jpg)

**Reply by Nllii** [reply]
MacOs Monterey should have no issues(as for now).

### Lastly, please ask the lovely people at  tigerbrew if they have some time to give, in starting a Catalina or Monterey branch https://github.com/mistydemeo/tigerbrew like they did 10 years ago.

**Comment by sroby** [comment]
Homebrew has always had this policy, it didn't "grow" out of nowhere.

**Reply by euguk007** [reply]
@sroby - my disappointment is not with the policy. It is with the absence of archives for past releases. Which, actually, is a common practice even for Apple.

**Reply by Bo98** [reply]
Homebrew operates as Git repositories. Every change we've ever made is archived that way.

**Reply by euguk007** [reply]
@Bo98 - Really? If all changes you make are archived, please assist me with finding the archives for Catalina bottles.
I would be most appreciative for your help here ;)

**Reply by Nllii** [reply]
Its a pain, but you can read the commit history and look for Catalina. 
Like so: 
sha256 catalina:Homebrew/homebrew-core created:2022
https://github.com/search?q=sha256+catalina%3AHomebrew%2Fhomebrew-core+created%3A2022&type=commits

**Comment by spaceshiptrooper** [comment]
I definitely agree with you on this. I can't use newer macOS versions because I use the Adobe CS6 suits and I can't get off of them because I specifically use Fireworks which only has 32bit. The last supported version of macOS for running 32bit is Mojave. So I'm stuck on Mojave for that reason. PHP 8.1.13 using Homebrew was working for me last week. Then I did an upgrade which then upgraded the dependencies broke my PHP upgrade installation, even though I have another PHP installation on another macOS that's running PHP 8.2 on Mojave as well. So whatever they upgraded for those dependencies are now broken since the last update.

I'm currently getting this error now when trying to install PHP's dependencies after it broke from the upgrade.

```
==> ./configure --prefix=/usr/local/Cellar/freetype/2.13.0 --enable-freetype-config --without-harfbuzz
==> make
Last 15 lines from /Users/{REDACTED}/Library/Logs/Homebrew/freetype/02.make:
libtool: compile:  clang -pedantic -std=c99 -I/private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/src/pcf -I/private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/objs -I./builds/unix -I/private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/include -c -Wall -g -O2 -fvisibility=hidden -DDARWIN_NO_CARBON -I/usr/local/Cellar/libpng/1.6.39/include/libpng16 -pthread "-DFT_CONFIG_CONFIG_H=<ftconfig.h>" "-DFT_CONFIG_MODULES_H=<ftmodule.h>" "-DFT_CONFIG_OPTIONS_H=<ftoption.h>" -DFT2_BUILD_LIBRARY /private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/src/pcf/pcf.c  -fno-common -DPIC -o /private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/objs/.libs/pcf.o
libtool: compile:  clang -pedantic -std=c99 -I/private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/src/bdf -I/private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/objs -I./builds/unix -I/private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/include -c -Wall -g -O2 -fvisibility=hidden -DDARWIN_NO_CARBON -I/usr/local/Cellar/libpng/1.6.39/include/libpng16 -pthread "-DFT_CONFIG_CONFIG_H=<ftconfig.h>" "-DFT_CONFIG_MODULES_H=<ftmodule.h>" "-DFT_CONFIG_OPTIONS_H=<ftoption.h>" -DFT2_BUILD_LIBRARY /private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/src/bdf/bdf.c  -fno-common -DPIC -o /private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/objs/.libs/bdf.o
libtool: compile:  clang -pedantic -std=c99 -I/private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/src/sfnt -I/private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/objs -I./builds/unix -I/private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/include -c -Wall -g -O2 -fvisibility=hidden -DDARWIN_NO_CARBON -I/usr/local/Cellar/libpng/1.6.39/include/libpng16 -pthread "-DFT_CONFIG_CONFIG_H=<ftconfig.h>" "-DFT_CONFIG_MODULES_H=<ftmodule.h>" "-DFT_CONFIG_OPTIONS_H=<ftoption.h>" -DFT2_BUILD_LIBRARY /private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/src/sfnt/sfnt.c  -fno-common -DPIC -o /private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/objs/.libs/sfnt.o
In file included from /private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/src/sfnt/sfnt.c:21:
/private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/src/sfnt/pngshim.c:411:7: error: expected expression
      FALL_THROUGH;
      ^
/private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/include/freetype/internal/compiler-macros.h:46:27: note: expanded from macro 'FALL_THROUGH'
#    define FALL_THROUGH  __attribute__(( __fallthrough__ ))
                          ^
1 error generated.
make: *** [/private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/objs/sfnt.lo] Error 1
make: *** Waiting for unfinished jobs....
libtool: compile:  clang -pedantic -std=c99 -I/private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/src/pcf -I/private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/objs -I./builds/unix -I/private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/include -c -Wall -g -O2 -fvisibility=hidden -DDARWIN_NO_CARBON -I/usr/local/Cellar/libpng/1.6.39/include/libpng16 -pthread "-DFT_CONFIG_CONFIG_H=<ftconfig.h>" "-DFT_CONFIG_MODULES_H=<ftmodule.h>" "-DFT_CONFIG_OPTIONS_H=<ftoption.h>" -DFT2_BUILD_LIBRARY /private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/src/pcf/pcf.c -o /private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/objs/pcf.o >/dev/null 2>&1
libtool: compile:  clang -pedantic -std=c99 -I/private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/src/bdf -I/private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/objs -I./builds/unix -I/private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/include -c -Wall -g -O2 -fvisibility=hidden -DDARWIN_NO_CARBON -I/usr/local/Cellar/libpng/1.6.39/include/libpng16 -pthread "-DFT_CONFIG_CONFIG_H=<ftconfig.h>" "-DFT_CONFIG_MODULES_H=<ftmodule.h>" "-DFT_CONFIG_OPTIONS_H=<ftoption.h>" -DFT2_BUILD_LIBRARY /private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/src/bdf/bdf.c -o /private/tmp/freetype-20230215-4853-17tv6ao/freetype-2.13.0/objs/bdf.o >/dev/null 2>&1

Do not report this issue to Homebrew/brew or Homebrew/core!
```

These are the dependencies that are broken and have the same error as the freetype one above.

`freetype, fontconfig, jpeg-xl, aom, libavif, gd`

I can't even go back and compile from source anymore because I don't have those bottles or formulas and can't compile them through Homebrew either.

----

EDIT: Looks like I was able to fix this by going to the working PHP installation with PHP 8.2.1 and take the files from there. So what I did was go into `/usr/local/Cellar/freetype/2.12.1` and pasted that folder into the broken PHP installation. Then I had to also add in the freetype.rb file found in `/usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula`. I tried to do a `brew install php` and freetype doesn't appear anymore in the list of dependencies. Will have to do that same process for the rest of them.

----

LAST EDIT: I got it to work.
![Screen Shot 2023-02-15 at 6 45 46 PM](https://user-images.githubusercontent.com/22463623/219229433-1579074a-4fc3-4193-b31c-9080a99b34bf.png)

So what I did was there were some things actually missing even after I had copied over those dependencies. First thing I was seeing was

```
Error: An exception occurred within a child process:
  RuntimeError: /usr/local/opt/freetype not present or broken
Please reinstall freetype. Sorry :(
```

After finding this out, all I had to do was create the symbolic links because there aren't there when I copied them over. Mind you, this is a fresh macOS installation. So for obvious reasons those symbolic links weren't there. All I did was `sudo ln -s /usr/local/Cellar/freetype /usr/local/opt/freetype`. And I did that for all of the dependencies that I had copied over.

After that, I kept getting this error when trying to install PHP.

```
checking for libjpeg... no
checking for libXpm... no
checking for FreeType 2... no
checking whether to enable JIS-mapped Japanese font support in GD... no
checking for gdlib >= 2.1.0... no
configure: error: Package requirements (gdlib >= 2.1.0) were not met:

No package 'gdlib' found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables GDLIB_CFLAGS
and GDLIB_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.

Do not report this issue to Homebrew/brew or Homebrew/core!

These open issues may also help:
php 8.2.3 https://github.com/Homebrew/homebrew-core/pull/123237
php@8.1 8.1.16 https://github.com/Homebrew/homebrew-core/pull/123238

Error: You are using macOS 10.14.
We (and Apple) do not provide support for this old version.
It is expected behaviour that some formulae will fail to build in this old version.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from MacHomebrew on Twitter.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.


Error: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences.

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 11.3.1.
```

So I immediately thought maybe I'm missing libjpeg. I tried installing libjpeg and it turns out, it's actually just the jpeg formula. So I did a `brew uninstall jpeg-turbo` and then got this error

```
Error: /usr/local/Cellar/libavif is not a valid keg
```

So I'm thinking to myself "what is going on? I have libavif already". So I tried to uninstall libavif and tried to reinstall it because I thought maybe the copied down version was having an issue. Got the same error again so I tried uninstalling it once more. Then I got this error

```
Error: No such keg: /usr/local/Cellar/libavif
```

After finding that out, I did `brew install libavif` and then came to this error.

```
--   Found libsharpyuv, version 1.3.0
-- libavif: libsharpyuv found; sharp rgb to yuv conversion enabled.
-- Checking for module 'aom'
--   No package 'aom' found
CMake Error at /usr/local/Cellar/cmake/3.25.2/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find aom (missing: AOM_LIBRARY AOM_LIBRARIES AOM_INCLUDE_DIR)
  (found version "")
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.25.2/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  cmake/Modules/Findaom.cmake:32 (find_package_handle_standard_args)
  CMakeLists.txt:416 (find_package)


-- Configuring incomplete, errors occurred!
See also "/tmp/libavif-20230215-71507-1mxvef8/libavif-0.11.1/build/CMakeFiles/CMakeOutput.log".

Do not report this issue to Homebrew/brew or Homebrew/core!
```

Shortly after, I found out that I was also missing another link for aom even though I have the files in there and have the symbolic links to the `/usr/local/opt` folder. So I did a `sudo find / -name "*aom*"` and found out that there indeed was a missing symbolic link to the include folder. So I did `sudo ln -s /usr/local/Cellar/aom/3.5.0_1/include/aom /usr/local/include/aom`.

After that, I tried `brew install libavif` and got the same error saying

```
-- Checking for module 'aom'
--   No package 'aom' found
```

Though this time it gave me more information at the bottom. Turns out, I was missing the link that Homebrew does for all of the dependencies I copied over. So all I needed to do was `brew link ____` and then reinstalled libavif. After that, I installed PHP and it worked perfectly.

So in all honesty, if I knew what I was missing, I could have cut down the process a lot more. tl;dr

1. Copy over dependency files from working PHP installation into `/usr/local/Cellar`.
2. Create symbolic link for dependencies by doing `sudo ln -s /usr/local/Cellar/{FORMULA} /usr/local/opt/{FORMULA}`.
3. Create missing symbolic link for aom by doing `sudo ln -s /usr/local/Cellar/aom/3.5.0_1/include/aom /usr/local/include/aom`.
4. Do `brew link {FORMULA}` to link all of the copied over dependencies.
5. Do `brew install php` and it should work.

---

EDIT: I have uploaded all of the files that I have for Homebrew to this [repo](https://github.com/spaceshiptrooper/homebrew).

**Reply by p-linnane** [reply]
>I can't use newer macOS versions because I use the Adobe CS6 suits and I can't get off of them because I specifically use Fireworks which only has 32bit. The last supported version of macOS for running 32bit is Mojave. So I'm stuck on Mojave for that reason.

I ask this in all sincerity, because I truly do not understand: how long do you think it will be tenable to keep using this OS/software combination? The last release of CS6 was almost 11 years ago, and Mojave has been unsupported for almost a year and a half. Would it not be easier to adapt to new technology instead of investing all this time creating workarounds for unpatchable software?

**Reply by sroby** [reply]
Honestly at this point, anyone dead set on still using discontinued software like CS6 that got killed during the Great 32-bit Purge but also has a Windows version, should just switch to Windows. As Macs that are old enough to run Mojave are gradually fading away, this is inevitable, your setup is already on borrowed time, you'll have to move on eventually.

Alternatively, if even that option is not acceptable at the present, I would suggest anyone who still needs Mojave to setup their Mac to dual-boot with a modern macOS, and switch to Mojave whenever they need to run 32-bit software. Dual-booting multiple macOSes is actually really easy since the switch to APFS, you can have both Mojave and current macOS share the same APFS container, and not have to worry about defining fixed amounts of storage space between the two system volumes. I've done it myself at one point, up until I was ready to ditch Mojave for good.

**Reply by GrowYourOwnFreedom** [reply]
can anyone help me? 
I have practically zero git experiance and i have a similar issue.
I'm trying to set up my mid 2012 mbp (for use on a coding bootcamp) so i can start learning how to use git/ github.
I have set up everything but when I try to use brew to download something it tries to update itself.(I only just updated from mavericks to catalina) 
That keeps  failing and when i try to install brew it says that because im on catalina my sytem is not supported and I should make a pull request instead of using offical channels to ask for help.
 Problem is I don't know how to sort it all out really as i dont really know how to use git to manually install different parts of software from old commits yet.
Why cant brew find the right commits itself and update to the last working version for my os?

**Reply by sroby** [reply]
My suggestion to you would be to use OpenCore Legacy Patcher to install Ventura on your mid-2012 MBP. According to the supported models page all 2012 MBPs are listed as "Everything is supported" so it should work well.
https://dortania.github.io/OpenCore-Legacy-Patcher/MODELS.html#macbook-pro

**Reply by spaceshiptrooper** [reply]
> > I can't use newer macOS versions because I use the Adobe CS6 suits and I can't get off of them because I specifically use Fireworks which only has 32bit. The last supported version of macOS for running 32bit is Mojave. So I'm stuck on Mojave for that reason.
> 
> I ask this in all sincerity, because I truly do not understand: how long do you think it will be tenable to keep using this OS/software combination? The last release of CS6 was almost 11 years ago, and Mojave has been unsupported for almost a year and a half. Would it not be easier to adapt to new technology instead of investing all this time creating workarounds for unpatchable software?

This isn’t a workaround though. It was supported even up til the error occurred. The reason why it’s no longer working is because the Homebrew team changed something in the configuration files which errored out. The issue isn’t “can I support new software on old hardware”, the issue is “if the Homebrew team breaks something in the configuration files, how can I go back to a working state”. I’m sure this error also occurs on newer versions of macOS as well. The only reason why you guys haven’t ran into it is because the packages aren’t being updated for you or you aren’t using these specific packages that are broken.

**Reply by spaceshiptrooper** [reply]
> Honestly at this point, anyone dead set on still using discontinued software like CS6 that got killed during the Great 32-bit Purge but also has a Windows version, should just switch to Windows. As Macs that are old enough to run Mojave are gradually fading away, this is inevitable, your setup is already on borrowed time, you'll have to move on eventually.
> 
> Alternatively, if even that option is not acceptable at the present, I would suggest anyone who still needs Mojave to setup their Mac to dual-boot with a modern macOS, and switch to Mojave whenever they need to run 32-bit software. Dual-booting multiple macOSes is actually really easy since the switch to APFS, you can have both Mojave and current macOS share the same APFS container, and not have to worry about defining fixed amounts of storage space between the two system volumes. I've done it myself at one point, up until I was ready to ditch Mojave for good.

This argument I have quoted from you contradicts the suggestion you gave to GrowYourOwnFreedom. Installing newer macOS versions on “unsupported” hardware has the same issues you’re trying to argue against. Newer macOS versions are “unsupported” on old hardware so there can always be some weird buggy side effect with installing new versions of macOS on old hardware. Especially if you’re using a 3rd party software to flash the new macOS version onto the old hardware and not the proper way of going through the official installer.

Lastly, the reason why I don’t agree with your argument is because this is just like situations where people throw out **good** hardware and buy new hardware because it’s the “hip” thing to do. You do realize you’re throwing out a device that cost you $1,299 in the start right? This is the reason why people rummage through garbage dumpsters, because apparently people have TOO much money to be throwing out good working hardware. That’s like throwing away your $70 K car because it made a “little” noise. Until it’s dead on its knees and it no longer is working, it should still be used or you’ll be wasting your money. Hence why I’m still using old hardware because my hardware is still running perfectly fine. Even more fine than brand spanking new Macs.

Lastly, these “workarounds” are only occurring now because the Homebrew team changed something in the files which broke the installation process. Check the error again, if you’re a developer, you’d know that the error relates to a defect the developer of that file did. So it “could” be intentional, but it could also be human error which broke the file. It doesn’t necessarily “have” to be a hardware/“unsupported” issue.

**Reply by sroby** [reply]
You seem to be conflating two separate situations. There is no contradiction to what I said, GrowYourOwnFreedom has not stated that they need to run an outdated macOS to run old software, and therefore it would make sense for them to use OCLP to install a modern macOS on their old MBP so they can keep using it with current software. Yes, OCLP is not perfect, but in the end it's still a better solution than running an outdated insecure macOS that might not run all current software.

**Reply by p-linnane** [reply]
> This isn’t a workaround though. It was supported even up til the error occurred. The reason why it’s no longer working is because the Homebrew team changed something in the configuration files which errored out. The issue isn’t “can I support new software on old hardware”, the issue is “if the Homebrew team breaks something in the configuration files, how can I go back to a working state”. I’m sure this error also occurs on newer versions of macOS as well. The only reason why you guys haven’t ran into it is because the packages aren’t being updated for you or you aren’t using these specific packages that are broken.

It is a workaround, as you are on an unsupported OS. The reasoning of "it was working so it was supported" does not hold up. [Our install guide](https://docs.brew.sh/Installation) states:
>macOS 11 (Big Sur) or higher is best and supported, 10.11 (El Capitan) – 10.15 (Catalina) are unsupported but may work and 10.10 (Yosemite) and older will not run Homebrew at all.

We, as volunteer maintainers, have a limited amount of time to work on Homebrew. We utilize this time to develop new features and support the OS's that Apple supports. We do not intentionally and maliciously break compatibility with older systems, but we can no longer devote time to troubleshooting for a subset of users. If something breaks for unsupported OS's through the course of normal development, that's just the way it is. If you [check our analytics for the last 90 days](https://formulae.brew.sh/analytics/os-version/90d/) you'll see that Mojave represents 0.82% of all installs during that period for users who are sending analytics. That same page will show you almost 94% of those users are on supported OS's. We have to prioritize our time.

I do want to say that I appreciate you posting solutions for your issue in this thread so that it can help others.

**Comment by GrowYourOwnFreedom** [comment]
Thanks!

Kind regards
  Jack
________________________________

**Comment by alextooter** [comment]
It become truelly "Missing".

**Comment by kencu** [comment]
I use several different package managers and different OS versions for different things... on my desk right now isa Windows system, an Ubuntu server installation, and a system running MacOSX 10.6.8. Each of these has unique attributes that makes the day go by more smoothly, and makes my life easier. You want to set up and run a server -- use Ubuntu. Anything else is really a make work project. You want to run some custom software to allow your printer to receive and send faxes -- that's a job for a PC and the custom software that comes with the printer from the manufacturer to do that. And for day to day use -- MacOS wins.

I have a few systems that run OCLP to upgrade them beyond what Apple allows -- and some of those run homebrew that way.

Where I am going with this is -- don't fight the current. Homebrew wants to keep cutting edge on their system version support for many reasons, all of them good reasons, and that has advantages. I have several systems running homebrew for just that reason. Homebrew works great for the current systems.

On the other hand, MacPorts has really outstanding support for older OS versions, and intends to keep this support going. I mean -- it's really very good. You can criticize the need for this, but -- there is a need. So if you want to run software on any older systems, look there, rather than fight the wind. It is easy to set up, easy to install, easy to maintain, and supports the older systems (as much as they can be supported, which is pretty good really) all the way back to 10.4 Tiger PPC. Yes, really.  It is not really a "ship it if it builds" situation -- things need to work -- but also passing 100% of the test suite is not a criteria for inclusion either, because frankly very little software can do that.

Use the tool that is made for the job.

**Comment by barracuda156** [comment]
Just use MacPorts, there is no point in trying to fix Homebrew.
